File: Find min in rotated array.java

import java.util.Scanner;

class Solution {
    public int findMin(int[] nums) {
        if(nums.length < 2)
            return nums[0];
        int left = 0, right = nums.length - 1;

        while(left <= right ){
            int mid = (left + right) / 2;
            if(nums[mid] >= nums[right])
                left = mid + 1;
            else
                right = mid;
        }
        return nums[right];
    }
    public static void main(String[] args) {
        Solution sln = new Solution();
        Scanner sc = new Scanner(System.in);
        int n;
        System.out.println("Enter the size of the array");
        n = sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter " + n + " elements");
        for(int i=0; i<n; i++)
            arr[i] = sc.nextInt();
        
        int res = sln.findMin(arr);

        System.out.println("Min element in the array is " + res);
        
    }
}

File: Find min in rotated array.py
class Solution:
    def findMin(self, arr):
        if len(arr) <= 1:
            return arr[0]
        l = 0
        r = len(arr) - 1
        while l <= r:
            m = (l+r) // 2

            if arr[m] < arr[m-1]:
                return arr[m]
            if arr[m]  >= arr[r]:
                l = m + 1
            else:
                r = m
        
obj = Solution();
arr = [5,1,2,3,4]
res = obj.findMin(arr)

print(res)